subctl/issue-89-reconstructing-broker-info.md

```diff
@@ -0,0 +1,70 @@
+# Recover broker-info.subm file from the running Broker
+
+Related Issue:
+[Option for recovering the broker-info.subm file from a running broker cluster](https://github.com/submariner-io/enhancements/issues/143)
+
+## Summary
+
+`broker-info.subm` file is created when a broker is installed. The file contains information critical to have a functional environment.
+There is a possibility that the file could be deleted or lost for one or the other reason. In such cases, no new clusters could be joined.
+
+This proposal describes a way to recover the `broker-info.subm` file from the running broker.
+
+## Proposal
+
+A new `subctl` command will be added that would fetch needed data from the broker and write it to `broker-info.subm` file. The file
+would be Base64 encoded.
+
+## Design details
+
+* subctl command name: `recover-brokerinfo`
+* Flags for the command:
+  * `submconfig` to specify kubeconfig of the cluster in which Submariner is installed
+  * `namespace` to specify the namespace in which broker is installed
+  * `ipsec-psk-from` to specify the file to read PSK secret from
+  * All flags that `restConfigProducer` provides
+* Data to be fetched from running broker:
+  * Broker URL
+  * Client token
+  * IPSec PSK
+  * Components installed i.e. Submariner and Service discovery
+  * whether service discovery component is installed
+  * default custom domains
+
+The last three information will be retrieved from the broker `Spec`, Broker URL from the user provided kubeconfig, client token from the
+Broker admin service account `submariner-k8s-broker-admin` and IPSec PSK with the following logic:
+
+```mermaid
```

Comment:
Review this under `rich diff` format (select from the top right button) to see the flowchart